src/crc32.c

```diff
@@ -100,7 +100,7 @@ static uint32_t crc32_calc(const uint32_t crc, const uint8_t data)
 	return (crc << 8U) ^ crc32_table[((crc >> 24U) ^ data) & 0xffU];
 }
 
-bool generic_crc32(target_s *const target, uint32_t *const result, const uint32_t base, const size_t len)
+static bool generic_crc32(target_s *const target, uint32_t *const result, const uint32_t base, const size_t len)
```

Comment:
If this is marked `inline`, does the compiler elide the call penalty `bmd_crc32()` should introduce? Or perhaps a better question: does adding the keyword make any chance to the resulting binary size? (same for stm32_crc32() below)

Reply:
Just `inline` breaks at link-time (after I un-static `crc32_calc` and `crc32_table` per GCC errors) with `src/crc32.c:220: undefined reference to 'generic_crc32'`. Similar error for `read_be4`, it being `static` also forces `stm32_crc32` impl to be `static` if the latter wanted to be `inline`. And even `static inline bool stm32_crc32()` results in this codegen:
```
0800550c <bmd_crc32>:
 800550c:       e786            b.n     800541c <stm32_crc32>
 800550e:       bf00            nop
```
i.e. `static inline` does not shrink, because it depends on the compiler weighting/decisionmaking. 
Adding `__attribute__((always_inline))` shrinks binary size back by 4, eliminating the cheap branch/jump on `native` and on `hosted`, but I don't trust toolchain-specific attributes to not cause FTBFS again.
At this point I could drop or keep these two lines of changes, it doesn't matter right now (but it could matter in the future when searching for optimization opportunities).

Reply:
So it's doing the thing we worried it would.. well, given the new direction this PR's taking, we'll mark this resolved.